pkg/ecdsa/tss/protocol_announce.go

```diff
@@ -44,9 +57,26 @@ func AnnounceProtocol(
 			case msg := <-announceInChan:
 				// Since broadcast channel has an address filter, we can
 				// assume each message come from a valid group member.
-				receivedMemberIDs[msg.SenderID.String()] = msg.SenderID
+				publicKey, err := msg.SenderID.PublicKey()
+				if err != nil {
+					logger.Errorf(
+						"could not get public key for member [%s] of keep [%v]: [%v]",
+						msg.SenderID.String(),
+						keepAddress,
+						err,
+					)
+					continue
+				}
 
-				if len(receivedMemberIDs) == membersCount {
+				memberAddress := "0x" + hex.EncodeToString(publicKeyToAddressFn(*publicKey))
```

Comment:
By adding `0x` we are tighting TSS implementation to ethereum specific prefixing.

Reply:
0x used here has nothing to do with Ethereum prefixing, it's how hexadecimals are [usually](https://stackoverflow.com/questions/2670639/why-are-hexadecimal-numbers-prefixed-with-0x) prefixed and because converter currently represents messages as byte array it makes sense. The approach abstracting address conversion presented in this PR is not a final solution. For example, it wouldn't work well with BTC addresses. It is a step forward but there is still work ahead of us. I do not think the address should be represented as a byte array eventually. I'd rather go with just a string.

Reply:
As you mentioned this solution this solution is not generic and won't work well for all chains. We will have to revisit it later.